### ISSUE TITLE
[Bug] Custom document as symlink is being resolved as real path

### ISSUE THREAD

**Issue by martynaskre** [hestiacp/hestiacp#2539]
### Describe the bug

Hestia always changes symlinked directory to actual directory.

For example, if custom document root is set to _/home/user/web/site.com/public_html/current/public_ Hestia automatically changes it to _/home/user/web/site.com/public_html/releases/12/public_.

### Tell us how to replicate the bug

1. Check checkbox "Custom document root"
2. Change document root to symlinked directory

### Which components are affected by this bug?

Backend Web Server (Nginx, Apache2)

### Hestia Control Panel Version

1.5.11

### Operating system

Ubuntu 20.04.4

**Comment by jaapmarcus** [comment]
```
        real_target_directory="$(readlink -e "${target_domain_directory}/public_html/$target_directory/")"
```

Probally the cause of it

**Comment by Quiesss** [comment]
> ```
>         real_target_directory="$(readlink -e "${target_domain_directory}/public_html/$target_directory/")"
> ```
> 
> Probally the cause of it

Does this problem have any solution?

**Comment by jaapmarcus** [comment]
Issue is do we call this "Bug" or a security feature

If we allow the user to set an dynamic link the user may create an symbolic out side his home folder after the dynamic links has been se.  

I don't think this is something we want to allow that.

**Comment by jaapmarcus** [comment]
> > ```
> >         real_target_directory="$(readlink -e "${target_domain_directory}/public_html/$target_directory/")"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Probally the cause of it
> 
> 
> Does this problem have any solution?

Solution would be create a custom template with the new docroot to the symlink and

**Comment by jaapmarcus** [comment]
I will close the issue for the reason provided here (https://github.com/hestiacp/hestiacp/issues/2539#issuecomment-1139417937)

**Comment by Quiesss** [comment]
> > > ```
> > >         real_target_directory="$(readlink -e "${target_domain_directory}/public_html/$target_directory/")"
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Probally the cause of it
> > 
> > 
> > Does this problem have any solution?
> 
> Solution would be create a custom template with the new docroot to the symlink and

Can you point to the manual or part of the documentation where custom templates are described, because I have already met a similar solution to this problem, but I could not figure it out yet. Thank you

**Comment by ScIT-Raphael** [comment]
@Quiesss https://docs.hestiacp.com/admin_docs/web.html#how-do-web-templates-work

**Comment by jaapmarcus** [comment]
https://docs.hestiacp.com/admin_docs/web.html

Example:
https://github.com/jaapmarcus/web-templates/blob/master/templates/apache2/shopware6.tpl

**Comment by LucaRed** [comment]
Here's the workaround I'm using if anybody stumbles here from a search.

My deployment script requires the docroot to be a symlink that always point to the latest release. I couldn't change the script and I didn't want to mess around with default templates.

Instead, I just added this command at the end of my deployment script (where `current` is the symlink name):

```php
sudo v-change-web-domain-docroot myuser example.com example.com current/public current/public
```

Since `myuser` is not a sudoer, I had to add this command in my sudoers file, like this:

```
myuser ALL=(ALL) NOPASSWD: /usr/local/hestia/bin/v-change-web-domain-docroot myuser example.com example.com current/public current/public
```

The command just forces HestiaCP to "reload" the real target directory based on the current symlink. I know it's not a perfect solution, but it works.
